modules/apps/mentions/mentions-service/src/main/java/com/liferay/mentions/internal/util/DefaultMentionsNotifier.java

```diff
@@ -205,7 +221,32 @@ private String[] _getMentionedUsersScreenNames(
 	@Reference
 	private PortletPermission _portletPermission;
 
+	private ServiceTrackerMap<String, MentionsMatcher> _serviceTrackerMap;
+
 	@Reference
 	private UserLocalService _userLocalService;
 
+	private class MentionsMatcherRegistry {
```

Comment:
You don't need to create a class here, just put the method `getMentionsMatcher` inside the `DefaultMentionsNotifier` class.